libraries/apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/ClientCacheExtensions.kt

```diff
@@ -121,28 +124,32 @@ fun ApolloClient.Builder.store(store: ApolloStore, writeToCacheAsynchronously: B
       .writeToCacheAsynchronously(writeToCacheAsynchronously)
 }
 
+@Deprecated(level = DeprecationLevel.ERROR, message = "Exceptions no longer throw", replaceWith = ReplaceWith("watch()"))
+@ApolloDeprecatedSince(v4_0_0)
+@Suppress("UNUSED_PARAMETER")
+fun <D : Query.Data> ApolloCall<D>.watch(
+    fetchThrows: Boolean,
+    refetchThrows: Boolean,
+) = watch()
```

Comment:
maybe throw instead? It's better to have a clear error than not knowing where to look?

Reply:
It's `DeprecationLevel.ERROR` so you shouldn't be able to call this, but I was thinking of the case where the code was compiled with 3.x but is running with 4.x. On the other hand this will surely not behave as expected in any case, so crashing early with a clear message may actually be better in fact 👍